### DISCUSSION TITLE
Free components render extra large!

### DISCUSSION THREAD

**Discussion by BaltWeb** [tailwindlabs/tailwindcss#11909] [Help]
I am new to Tailwind and would like to start by using the free of charge components, with code readily available to code and use from the Tailwind UI page.  Taking the first Application UI free component as example to explain the issue, this :

https://tailwindui.com/components/application-ui/application-shells/stacked

Coping and pasting the HTML code to render, on running it does not render correctly; and is displayed very large in the browser, please see P1 & P2 attached. 

This issue persists when running via the terminal / localhost and also when run with VS Code, Apache and Codepen. I also tested for this issue on a different pc using Codepen only; finding the same large rendering issue also. 

Further to the above, I have gone further and tried some of the other free components available, with the same rendering issue also!

Please let me know what may be the root cause of the issue and  how to resolve it?

**Comment by wongjn** [comment]
Consider providing a [Tailwind Play](https://play.tailwindcss.com) that reproduces the unexpected behavior.

**Reply by BaltWeb** [reply]
Thanks, run the HTML on Tailwind Play and it rendered perfectly; as shown in the Tailwind UI page. OK but where does this leave me? I need it to run in my VS Code; how do I get it to run in the same way on VS Code?

**Reply by wongjn** [reply]
VSCode is a code editor, and HTML does not typically "run" within it unless the HTML is displayed as a "special document" – could you please clarify what you mean by "run in my VS Code"?

**Reply by BaltWeb** [reply]
Preview in browser via an a extension in order to render the HTML.

**Reply by wongjn** [reply]
Consider checking:
- Whether HTML is being rendered at a larger zoom level.
- Whether the viewport meta tag has been added to the HTML page (and that the HTML engine that renders the page complies with the specification of this).
- Whether this could be a bug with the extension.

**Reply by BaltWeb** [reply]
First and second points checked and rechecked. 

Regarding the third as explained at the start of this discussion:

    " This issue persists when running via the terminal / localhost and also when run with VS Code, Apache and Codepen. I also tested for 
      this issue on a different pc using Codepen only; finding the same large rendering issue also.

      Further to the above, I have gone further and tried some of the other free components available, with the same rendering issue also! "

Today I continued my testing of another 2 free Tailwind UI components:

https://tailwindui.com/components/ecommerce/components/product-overviews
and
https://tailwindui.com/components/marketing/sections/cta-sections

On yet a 3rd PC, pasting the HTML code and adjusting according to Tailwind instructions where required, to obtain similar enlarged and / or incorrect rendering using Codepen to render the HTML code of both examples.

I will be raising this discussion to ISSUE level !!!

**Reply by wongjn** [reply]
Consider providing a remotely-hosted git repository or a CodePen that reproduces the unexpected behavior.

**Reply by adamwathan** [reply]
As @wongjn said please provide a reproduction or it’s impossible to help. It’s definitely something specific to your setup as it doesn’t happen for us and it’s never been reported by anyone else. Definitely happy to help but need more to work with 🙏

**Reply by BaltWeb** [reply]
Will carry out test as requested this evening and share the repository, Thanks

**Reply by BaltWeb** [reply]
Please find link to Codepen for the first free UI where I encountered this issue:

https://codepen.io/BaltWeb/pen/VwqaVxw

**Reply by wongjn** [reply]
You seem to be missing Tailwind in that CodePen. Here it is with only Tailwind CDN added, which seems to be visually as intended: https://codepen.io/Wongjn/pen/rNoeQoQ. Please note, Tailwind CDN is not recommended for production and is only used in the CodePen since we cannot run a Tailwind compilation there.

**Reply by BaltWeb** [reply]
As it should render as shown on Tailwind UI page:

![As should render shown on Tailwind ](https://github.com/tailwindlabs/tailwindcss/assets/112821753/a8621f0f-0d7e-466a-9e48-1c74380f8896)

3 screenshots of how is renders on Codepen and also on Apache and VS Code:
![TSL01](https://github.com/tailwindlabs/tailwindcss/assets/112821753/ec167dd9-7376-40a5-b107-95530a826a85)
![TSL02](https://github.com/tailwindlabs/tailwindcss/assets/112821753/8e555c07-435e-46f6-b08c-128ac366c536)
![TSL03](https://github.com/tailwindlabs/tailwindcss/assets/112821753/fa59555f-7e59-4e35-a6ff-d843ecbc4646)

**Reply by wongjn** [reply]
Please see my previous comment: https://github.com/tailwindlabs/tailwindcss/discussions/11909#discussioncomment-6867761. It seems you have not included Tailwind-generated CSS in the webpage.

**Reply by BaltWeb** [reply]
Please explain further, as there I may have missed the instruction for this in UI HTML code; as there is for the two classes to add in this example. Kindly show me this instruction is shown?

**Reply by wongjn** [reply]
Consider reading [the documentation to getting started](https://tailwindui.com/documentation#getting-set-up).

**Comment by thecrypticace** [comment]
If it's not zoom related and components are rendering larger than you expect chances are that your browser or editor preview has a larger-than-normal font size specified which is the basis for the `rem` CSS unit and affects the spacing, sizing, and other scales that Tailwind uses. Another scenario in which this can happen is you are overriding the font-size in your CSS on the `html` element with a size that is larger than `16px`. 

A great way verify either of these is to add `html { font-size: 16px }` to the CSS and see if the components render normal size.

If neither of those help a CodePen with screenshots of everything you are seeing (CodePen UI included) will probably be necessary to figure out what the problem is.
